src/stylize.ts

```diff
@@ -97,3 +101,30 @@ export function nominalColorScaleForHighCardinality(specM: SpecQueryModel, schem
 
   return specM;
 }
+
+export function xAxisOnTopForHighYCardinalityWithoutColumn(specM: SpecQueryModel, schema: Schema): SpecQueryModel {
+ [Channel.COLUMN, Channel.X, Channel.Y].forEach((channel) => {
+        encQIndex[channel] = specM.getEncodingQueryByChannel(channel);
+      });
+
+  if (encQIndex[Channel.COLUMN] === undefined) {
+    const xEncQ = encQIndex[Channel.X];
+    const yEncQ = encQIndex[Channel.Y];
+    if (yEncQ !== undefined && yEncQ.field && contains([Type.NOMINAL, Type.ORDINAL], yEncQ.type)) {
```

Comment:
I think what you want to `scaleType === ScaleType.ORDINAL` instead of `contains([Type.NOMINAL, Type.ORDINAL], yEncQ.type)` which includes timeUnit that uses ordinal scale too.